### ISSUE TITLE
[MNT] show coverage corresponding to `test all` workflow on `README`

### ISSUE THREAD

**Issue by yarnabrina** [sktime/sktime#6035]
Currently README shows the code coverage on main branch corresponding to last commit. But this does not run all unit tests by default to keep regular CI faster, both pre and post PR. These values are pretty low, and the badge uses `https://img.shields.io/codecov/c/github/sktime/sktime?label=codecov&logo=codecov` as badge link.

![](https://img.shields.io/codecov/c/github/sktime/sktime?label=codecov&logo=codecov)

On the other hand `test all` workflow runs almost all unit tests (except datasets) as a CRON job every Sunday, and this leads to a more relevant coverage. This can be shown in the badge using `https://img.shields.io/codecov/c/github/sktime/sktime?label=codecov&logo=codecov&flag=complete` as link.

![](https://img.shields.io/codecov/c/github/sktime/sktime?label=codecov&logo=codecov&flag=complete)

The issue with this is it's not available for every single commit, so it'll appear as unknown very often. If we can find out to refer to the latest coverage corresponding to `test all` workflow (corresponding to `complete` flag), we can switch to this badge.

**Comment by fkiraly** [comment]
Duplicate or significant intersection with https://github.com/sktime/sktime/issues/5090 ?

Maybe let's close down the older one, as that does not suggest a solution.

**Comment by fkiraly** [comment]
is this a good first issue in your opinion, @yarnabrina?

**Comment by yarnabrina** [comment]
I don't think so, it may be very hacky and specific. Plus, it may need some quick merge on main to check if it even works, as other branches won't have coverage reports.

If it's okay with you, I'd loke to try this tonight and will request a quick merge if possible. 

https://docs.codecov.com/docs/carryforward-flags

**Comment by fkiraly** [comment]
sure. Btw if you can think of other "good first issues" for new contributors (therer are a lot! and of different py experience levels), it woudl be appreciate if you just add the label here adn there.

**Comment by yarnabrina** [comment]
The PR not work, and made things worse. Now complete flag link shows no badge. Opening a revert PR, but not sure how to fix this. Can we ask someone from `codecov` team if you know who to reach out to?

**Comment by fkiraly** [comment]
ouch. Any ideas what went wrong?

> Can we ask someone from `codecov` team if you know who to reach out to?

Not really, sorry.

**Comment by yarnabrina** [comment]
@MEMEO-PRO @Xinyu-Wu-0000 @duydl since you have suggested about GHA in different PR's, wondering if you can suggest anything regarding this?

**Comment by duydl** [comment]
Could it be because of https://docs.codecov.com/docs/codecov-yaml#expired-reports . There is no coverage data for the complete flag now https://app.codecov.io/gh/sktime/sktime?flags%5B0%5D=complete so it appeared as unknown.

**Comment by yarnabrina** [comment]
https://app.codecov.io/gh/sktime/sktime/flags?flags%5B0%5D=complete&flags%5B1%5D=complete

I am able to see previous coverage against complete flag, can you please check again? We have a CRON job every Sunday, so it's not expected to be empty.

**Comment by duydl** [comment]
There is historical coverage data recorded for all flags, but the report may be expired.

**Comment by yarnabrina** [comment]
The way (new) CI setup is that all PR's upload with tags identiying component name, python version and operating system along with incomplete, and test all does component, python, OS and complete. The scope of this issue is specifically for complete tag itself, which we'd like to use for the README badge.

(That said, most other tags missing coverage reports, e.g. classification on your image is worrying and completely unexpected to me. Any ideas @fkiraly / @benHeid ?)

Now regarding expired reports, can you please explain what exactly are these? Reading that page made me think if a report is uploaded that was originally created long time back (definition of long is to be configured). It doesn't seem like our case, so may be I am not following your point. Can you please elaborate?

**Comment by duydl** [comment]
> Can you please elaborate?

codecov merge the reports for different workflows to create the coverage stats. However, according to the https://docs.codecov.com/docs/codecov-yaml#expired-reports, report that is more than 12 hours old will be rejected by default. 

The case might also be that reports from test_all workflow (and other with different flags) were simply overwritten with newer report, as there is only one sktime report showing despite selecting all flags. 

btw, I think I filtered the complete flag in the gif, though the result would be the same for classification falg.

**Comment by yarnabrina** [comment]
> btw, I think I filtered the complete flag in the gif, though the result would be the same for classification falg.

This is what I see

![image](https://github.com/sktime/sktime/assets/39331844/91e1fa3d-74fb-445e-8a9d-8a8e36e9fb89)

> codecov merge the reports for different workflows to create the coverage stats. However, according to the https://docs.codecov.com/docs/codecov-yaml#expired-reports, report that is more than 12 hours old will be rejected by default.

I think there is some misunderstanding. We are not facing issues to publish our coverage reports (see the link I shared above and the screenshot). The reason why you are seeing empty from home page is there is no upload with `complete` flag for every single commit in `main` (it gets uploaded only for test all workflow, not for other workflows). This issue (#6035) is to plan how can we show the coverage corresponding to those runs of test all workflows only, as that runs the full test suite unlike that on PR runs. Does this make sense? If not, please let me know.

**Comment by yarnabrina** [comment]
Okay, the theory sounds fine, but I think it may not be right. There are many repositories that report coverage scores on their README's but last CI was run long back. For example, if you see this [dummy repo of mine](https://github.com/yarnabrina/learn-python-packaging), it shows coverage as 93% but CI was last ran over a month ago.

But it's possible that the expiry period theory is applicable only if all flags are not present in all commits. I don't know how to test that setup.

**Comment by duydl** [comment]
I think it works a little differently. The data coverage will remain of the latest commit. The expired period refers to which reports from individual workflow will be merged into the overall reports. All the flags related to individual workflow reports merged into the overall reports will be available, and return unknown badge otherwise. 
https://docs.codecov.com/docs/merging-reports

**Comment by duydl** [comment]
Based on the explanation above, preventing all but builds with complete flag from merging into master reports would likely be a better solution. We could still observe the coverage of incomplete tests. 

https://docs.codecov.com/docs/flags#hide-builds-eg-nightly-builds
